### ISSUE TITLE
AI Hub Support in Terraform

### ISSUE THREAD

**Issue by deeiip** [terraform-providers/terraform-provider-google#4708]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Google cloud added the the [AI Hub in Beta](https://cloud.google.com/ai-hub/). Currently in Terraform we can create a Ai Hub model but we can not create Ai Hub jobs, notebooks resources.
### New or Affected Resource(s)
[google_ai_hub ](https://cloud.google.com/ai-hub/)

* google_ai_hub
* google ml_engine

### References
* [google_ai_hub](https://cloud.google.com/ai-hub/)

**Comment by ndmckinley** [comment]
It's not immediately clear that this would be doable with our standard REST approach - it seems like we'd need to call out to Python, from the docs.  That might be hard to package.  If this acquires a lot of :+1: we'll consider that - otherwise, we most likely want to stick with REST-accessible resources.

**Comment by deeiip** [comment]
@ndmckinley Thanks! However I'm not very sure what I'm missing But I feel we might be able to manage it via Rest API (i.e. via service: [ml.googleapis.com](https://cloud.google.com/ml-engine/reference/rest/?apix=true) ).
Resource Jobs: [Rest API ref](https://cloud.google.com/ml-engine/reference/rest/v1/projects.jobs)
Resource Model: [Rest API ref](https://cloud.google.com/ml-engine/reference/rest/v1/projects.models)
Resource Model.version:  [Rest API ref](https://cloud.google.com/ml-engine/reference/rest/v1/projects.models.versions)

**Comment by ndmckinley** [comment]
We could definitely support the REST objects, but to upload the scikit models or tensorflow models, we'd need to call out to python, it seems.  I'm talking about the operations described [here](https://cloud.google.com/ml-engine/docs/deploying-models) - those are the ones that would be hard to support.  Would the REST objects be useful to you if we didn't have those operations?

**Comment by deeiip** [comment]
Yes It'd be definitely helpful in general sense. The way I understand it,
* ML jobs creates a ML model (e.g. `tensorflow_model.pb `or `simple_model.pkl`).  Now any custom python module used in the process is generally not part of infra and should not be a responsibility of Terraform.
To give an example: For [Dataflow Job](https://www.terraform.io/docs/providers/google/r/dataflow_job.html) anything pointed via [template_gcs_path](https://www.terraform.io/docs/providers/google/r/dataflow_job.html#template_gcs_path) is user code.
* In the same way when creating [models](https://cloud.google.com/ml-engine/reference/rest/v1/projects.models) or [models.version](https://cloud.google.com/ml-engine/reference/rest/v1/projects.models.versions) any user code (e.g. python modules) i.e. code pointed via `packageUris[]` in [projects.models.versions](https://cloud.google.com/ml-engine/reference/rest/v1/projects.models.versions#Version) is separate and should not be a part of Infra itself.

**Comment by noamt** [comment]
I'd be happy to help with this feature

**Comment by ndmckinley** [comment]
Awesome, all yours!  Let us know if we can be of any support here.  :)

**Comment by noamt** [comment]
It looks like magic modules already has the definition for models and model versions, which means it's technically possible to publish new versions.
So what's missing? Is it only a question of generating the terraform provider?

**Comment by deeiip** [comment]
We can always create new model+version. We can create Jobs, Notebooks etc.

**Comment by noamt** [comment]
Ah, OK.
It also looks like jobs are on the way:
https://github.com/GoogleCloudPlatform/magic-modules/pull/2565
